cmd/internal/textviewer.go

```diff
@@ -0,0 +1,53 @@
+package internal
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+func ViewInEditor(text string) error {
+	editor := os.Getenv("EDITOR")
+	if editor == "" {
+		return errors.New("Please define a text editor to use with the $EDITOR environment variable")
+	}
+
+	tmpfile, err := os.CreateTemp("", "prom-scrape-analyzer-*.txt")
+	defer func() {
+		if tmpfile != nil {
+			_ = tmpfile.Close()
+			_ = os.Remove(tmpfile.Name())
+		}
+	}()
+	if err != nil {
+		return errors.Wrap(err, "failed to create temporary file to display text")
+	}
+
+	_, err = tmpfile.WriteString(text)
+	if err != nil {
+		return errors.Wrap(err, "failed to write text to temporary file")
+	}
+
+	args := strings.Split(editor, " ")
+	args = append(args, tmpfile.Name())
+
+	cmd := exec.Command(args[0], args[1:]...)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+
+	var stderr bytes.Buffer
+	cmd.Stderr = &stderr
+
+	err = cmd.Run()
```

Comment:
Something is funky with this part and MacOS. It simply does not work on my local. I was thinking we can maybe open vim and quit the program for now.

Reply:
Ah okay, that sounds good, I figure debugging env specific issues would be time consuming. Looking at k9s code code for editing resources might help in the future.

 What was your `$EDITOR` command btw? And is it working with the changes you made?

I'm also on MacOS with an arm64 chip

Reply:
I tested `EDITOR=nvim` and `EDITOR=code`, because we drop the file right after we open the editor, sometimes the editor wont be able to read it. I left a comment about that as well.